### PR TITLE
Fixed issue #3263: Compare parent revision with working tree

### PR DESCRIPTION
**Pull request by vitaliyg2**
Implemented feature described in issue
["Compare parent revision with working tree"](https://gitlab.com/tortoisegit/tortoisegit/issues/3263)

### CODE REVIEW

**Comment by YueLinHo** [comment]
Thanks for your PR. :)

I am here for reminding you some:

1. The **subject of the commit** should be:  
    ```
    Fixed issue #3263: Compare parent revision with working tree
    ```
    For example:  
    image
    BTW, if you create a MR on GitLab, the issue will be automatically closed after the MR is merged. So, it would be great that creating the MR there.  
2. Modify the **`/src/Changelog.txt`** in the same commit
    Suppose the issue is a feature. So, append a new line "Fixed issue ... " to the `== Features ==` section.
    For example:  
    image

**Comment by vitaliyg2** [comment]
Hi, thanks for instructions,
I've updated commit subject and added a line to changelog.txt file.

**Comment by csware** [comment]
Thanks for this PR, however, I still do not see why this is necessary. One can easily select "Compare with working tree" and then append "~1" to version 1.

**Comment by vitaliyg2** [comment]
> 
> 
> Thanks for this PR, however, I still do not see why this is necessary. One can easily select "Compare with working tree" and then append "~1" to version 1.

Hi,

this is the feature that I was looking for in TortoiseGit some time ago and couldn't find it.
There is already very similar menu item that compares file with changes in selected commit with WC,
but there is no menu item to compare file before changes in current commit with WC.

Using "Compare with working tree" and appending ~1 is not that obvious and I didn't knew that this is possible at the time,
I think this requires more advanced understanding of how git and TortoiseGit are working and this might not be the case for significant amount of users.

File: src/Git/GitStatusListCtrl.cpp [hunk 1]
```diff
@@ -2764,7 +2763,22 @@ void CGitStatusListCtrl::StartDiffWC(int fileindex)
 	CTGitPath file1 = *ptr;
 	file1.m_Action = 0; // reset action, so that diff is not started as added/deleted file; see issue #1757
 
-	CGitDiff::Diff(GetParentHWND(), &file1, &file1, GIT_REV_ZERO, m_CurrentVersion.ToString(), false, false, 0, !!(GetAsyncKeyState(VK_SHIFT) & 0x8000));
+	CString version;
+	if (parent)
```
Comment (csware) [inline src/Git/GitStatusListCtrl.cpp]: Wouldn't it be easier to use `m_currentVersion.ToString() + L"~1"`?

Comment (vitaliyg2) [reply src/Git/GitStatusListCtrl.cpp]: "~1" allows to get parent commit for the first parent, and existing code allows to retrieve also other parents.
Code for evaluating parent commit was already implemented in "Revert to parent revision" action, I only did some refactoring to reuse it.

**Pull request merged**